app/controllers/admin_controller.rb

```diff
@@ -8,28 +8,118 @@ def index
+
+    # Validate pagination params
+    entries = permitted_params[:entries]&.to_i || 25
+    page = permitted_params[:page]&.to_i || 0
+    return head :bad_request unless entries >= 0 && page >= 0
+
+    # Validate sort params
+    order_by = permitted_params[:orderBy]
+    return head :bad_request unless order_by.nil? || order_by.blank? || %w[id email jurisdiction_path].include?(order_by)
+
+    sort_direction = permitted_params[:sortDirection]
+    return head :bad_request unless sort_direction.nil? || sort_direction.blank? || %w[asc desc].include?(sort_direction)
+    return head :bad_request unless (!order_by.blank? && !sort_direction.blank?) || (order_by.blank? && sort_direction.blank?)
+
+    # Get all users within the current user's jurisdiction
+    users = User.all.where(jurisdiction_id: current_user.jurisdiction.subtree_ids).joins(:jurisdiction).select(
+      'users.id, users.email, users.api_enabled, users.locked_at, users.authy_id, users.failed_attempts, jurisdictions.path '
+    )
+
+    # Filter by search text
+    users = filter(users, search)
+
+    # Sort
+    users = sort(users, order_by, sort_direction)
+
+    # Paginate
+    users = users.paginate(per_page: entries, page: page + 1)
+
+    # Get total count
+    total = users.total_entries
+
+    linelist = []
+    users.each do |user|
+      details = {
+        id: user.id,
+        email: user.email,
+        jurisdiction_path: user.path || '',
+        role: user.roles[0].name.split('_').map(&:capitalize).join(' ') || '',
+        is_locked: !user.locked_at.nil? || false,
+        is_API_enabled: user[:api_enabled] || false,
+        is_2FA_enabled: user.authy_id.nil? || false,
+        num_failed_logins: user.failed_attempts
+      }
+
+      linelist << details
+    end
+
+    render json: { linelist: linelist, total: total }
+  end
+
+  def sort(users, order_by, sort_direction)
+    return users if order_by.nil? || order_by.empty? || sort_direction.nil? || sort_direction.blank?
+
+    # Satisfy brakeman with additional sanitation logic
+    dir = sort_direction == 'asc' ? 'asc' : 'desc'
+
+    puts "order by: #{order_by}"
+    case order_by
+    when 'id'
+      users = users.order(id: dir)
+    when 'email'
+      users = users.order(email: dir)
+    when 'jurisdiction_path'
+      users = users.order(path: dir)
+    when 'num_failed_logins'
+      users = users.order(failed_attempts: dir)
+    end
+
+    users
+  end
+
+  def filter(users, search)
+    return users if search.nil? || search.blank?
+
+    users.where('users.id like ?', "#{search}%").or(
+      users.where('users.email like ?', "#{search}%")
+    )
+  end
+
   def create_user
-    permitted_params = params[:admin].permit(:email, :jurisdiction, :role_title)
-    roles = Role.pluck(:name)
+    permitted_params = params[:admin].permit(:email, :jurisdiction, :role, :is_API_enabled)
     email = permitted_params[:email].strip
     raise 'EMAIL must be provided' unless email
```

Comment:
We need to remove all of these `raise` directives in this controller, and instead return an applicable status code.

Reply:
Done and replaced with return `head :bad_request unless..`